src/IO/H5/VolumeData.cpp

```diff
@@ -367,6 +702,138 @@ void VolumeData::write_volume_data(
+
+    for (size_t element_index = 0;
+         element_index < block_number_for_each_element.size();
+         ++element_index) {
+      auto element_mesh = mesh_for_grid<SpatialDim>(
+          grid_names[element_index], grid_names, extents, bases, quadratures);
+      auto element_logical_coordinates_tensor =
+          logical_coordinates(element_mesh);
+
+      std::vector<std::array<double, SpatialDim>> element_logical_coordinates;
+      element_logical_coordinates.reserve(
+          element_logical_coordinates_tensor.get(0).size());
+
+      for (size_t k = 0; k < element_logical_coordinates_tensor.get(0).size();
+           ++k) {
+        std::array<double, SpatialDim> logical_coords_element_increment = {};
+        for (size_t l = 0; l < SpatialDim; l++) {
+          gsl::at(logical_coords_element_increment, l) =
+              element_logical_coordinates_tensor.get(l)[k];
+        }
+        element_logical_coordinates.push_back(logical_coords_element_increment);
+      }
+
+      auto block_logical_coordinates =
+          generate_block_logical_coordinates<SpatialDim>(
+              element_logical_coordinates, grid_names[element_index],
+              h_ref_per_block[block_number_for_each_element[element_index]]);
+
+      // Stores (B#, grid_point_coord_array) -> grid_point_number in an
+      // unordered_map and grid_point_coord_array by block
+      for (size_t k = 0; k < block_logical_coordinates.size(); ++k) {
+        std::pair<size_t, std::array<double, SpatialDim>> block_and_grid_point(
+            block_number_for_each_element[element_index],
+            block_logical_coordinates[k]);
+        block_and_grid_point_map.insert(
+            std::pair<std::pair<size_t, std::array<double, SpatialDim>>,
+                      size_t>(block_and_grid_point, grid_point_number));
+        grid_point_number += 1;
+
+        block_logical_coordinates_by_block
+            [block_number_for_each_element[element_index]]
+                .push_back(block_logical_coordinates[k]);
+      }
+    }
+
+    std::vector<int> new_connectivity;
+    new_connectivity.reserve(total_expected_connectivity);
+
+    for (size_t j = 0; j < block_logical_coordinates_by_block.size(); ++j) {
+      auto block_number = j;
+      auto connectivity_of_keys = generate_new_connectivity<SpatialDim>(
+          block_logical_coordinates_by_block[j], block_number);
+      for (const std::pair<size_t, std::array<double, SpatialDim>>& it :
+           connectivity_of_keys) {
+        new_connectivity.push_back(block_and_grid_point_map[it]);
+      }
+    }
+
+    // Deletes the existing connectivity and replaces it with the new one
+    const std::string path = "ObservationId" + std::to_string(obs_id);
+    detail::OpenGroup observation_group(volume_data_group_.id(), path,
+                                        AccessType::ReadWrite);
+    const hid_t group_id = observation_group.id();
+    H5Ldelete(group_id, "connectivity", H5P_DEFAULT);
```

Comment:
Can you move this into the `src/IO/H5/Helpers.cpp` file next to the function to write it? I realize it's trivial code right now, but if we decide to compress the connectivity in the H5 file, then the property list might not be `H5P_DEFAULT` anymore, and this could lead to difficult to find bugs. Then you can also use `CHECK_H5(H5L...)` to verify the operation succeeded.

Reply:
When moving the `H5Ldelete(...)` part into the file you specified, do I need to change `H5P_DEFAULT` to something else more general?

Reply:
The code there uses our own `h5p_default()` function, so just use that. Some of the HDF5 macro stuff causes a lot of warnings, so we wrap the code and silence warnings inside wrappers.